client/components/selected-image.jsx

```diff
@@ -1,20 +1,81 @@
 import React, { useState, useEffect } from 'react';
 
-export default function SelectedImage({ imageId }) {
+export default function SelectedImage({ imageId, user }) {
   const [image, setImage] = useState('');
+  const [isLiked, setIsLiked] = useState(false);
 
+  // Fetch the selected image's row from the database
   useEffect(() => {
     fetch(`/images/${imageId}`)
       .then(response => response.json())
-      .then(data => setImage(data));
+      .then(data => setImage(data))
+      .catch(error => console.error(error));
   }, [imageId]);
 
+  // Fetch a list of all liked images and check to see if the
+  // selected image is liked by the current user
+  useEffect(() => {
+    fetch(`/images/${imageId}/likedImage`)
+      .then(response => response.json())
+      .then(data => {
+        for (let i = 0; i < data.length; i++) {
```

Comment:
You could have written this with `data.some()` a little more succinctly, but this isn't a bad solution.